src/views/DatasetManager/components/DatasetList/DatasetList.js

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useCallback, useState } from 'react';
+import { Box, Divider, IconButton, List, ListItem, ListItemButton, ListItemText, Typography } from '@mui/material';
+import { DeleteForever as DeleteForeverIcon, Refresh as RefreshIcon } from '@mui/icons-material';
+import { DontAskAgainDialog } from '@cosmotech/ui';
+import { CreateDatasetButton } from '../../../../components';
+import { DatasetsUtils } from '../../../../utils';
+import { useTranslation } from 'react-i18next';
+import { useDatasetList } from './DatasetListHook';
+import { TwoActionsDialogService } from '../../../../services/twoActionsDialog/twoActionsDialogService';
+
+export const DatasetList = () => {
+  const { t } = useTranslation();
+  const [isRefreshConfirmationDialogOpen, setIsRefreshConfirmationDialogOpen] = useState(false);
+  const { sortedDatasetList, currentDataset, selectDataset } = useDatasetList();
+  const refreshDialogLabels = {
+    title: t('commoncomponents.datasetmanager.dialogs.refresh.title', 'Overwrite data?'),
+    body: t(
+      'commoncomponents.datasetmanager.dialogs.refresh.body',
+      'Your data will be lost, replaced with the imported one.'
+    ),
+    cancel: t('commoncomponents.datasetmanager.dialogs.cancel', 'Cancel'),
+    confirm: t('commoncomponents.datasetmanager.dialogs.refresh.overwriteButton', 'Overwrite'),
+    checkbox: t('commoncomponents.datasetmanager.dialogs.refresh.checkbox', 'Do not ask me again'),
+  };
+  const askConfirmationToDeleteDialog = useCallback(
+    async (event, datasetId) => {
+      event.stopPropagation();
+      const initialDatasetName = sortedDatasetList.find((dataset) => dataset.id === datasetId)?.name;
+      const datasetsToDelete = DatasetsUtils.getDatasetTree(datasetId, sortedDatasetList);
```

Comment:
Could we rename this const to `childrenToDelete` instead of `datasetsToDelete`? It makes it explicit that the initial dataset is not included in the list